File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/data/generators/trade-parameters.ts
import { objectName, volumeName } from './constants';

/** The trade related input parameters for the waveform generators. */
export class TradeParameters {
    private static readonly defaultVolume: number = 100;

    /** The volume, which is the same for all trades; should be positive. */
    volume: number = TradeParameters.defaultVolume;

    constructor(data?: TradeParameters) {
        if (data) {
            for (const property in data) {
                if (data.hasOwnProperty(property)) {
                    (<any>this)[property] = (<any>data)[property];
                }
            }
        }
    }

    toJSON(data?: any) {
        data = typeof data === objectName ? data : {};
        data[volumeName] = this.volume;
        return data;
    }
}

File: Mbs/Trading/Data/Generators/TradeParameters.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using Mbs.Trading.Data.Entities;

namespace Mbs.Trading.Data.Generators
{
    /// <summary>
    /// The <see cref="Trade"/> related input parameters for the waveform generators.
    /// A waveform generator produces trades with a constant volume.
    /// </summary>
    public class TradeParameters : IValidatableObject
    {
        /// <summary>
        /// Gets or sets the value of the volume, which is the same for all trades; should be positive.
        /// </summary>
        [Required(ErrorMessage = ErrorMessages.FieldIsRequired, AllowEmptyStrings = false)]
        [DefaultValue(DefaultParameterValues.TradeVolume)]
        [Range(0.0, double.MaxValue, ErrorMessage = ErrorMessages.FieldMustBePositive)]
        public double Volume { get; set; } = DefaultParameterValues.TradeVolume;

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}
